ledger/apply/application_test.go

```diff
@@ -1096,20 +1099,42 @@ func TestAppCallApplyDelete(t *testing.T) {
 	a.Equal(cbr, br)
 	a.Equal(basics.EvalDelta{}, ad.EvalDelta)
 
-	// check deletion on empty balance record - happy case
-	b.pass = true
-	b.balances[sender] = basics.AccountData{}
-	err = ApplicationCall(ac, h, &b, ad, &ep, txnCounter)
-	a.NoError(err)
-	a.Equal(appIdx, b.deAllocatedAppIdx)
-	a.Equal(1, b.put)
-	br = b.balances[creator]
-	a.Equal(cbr, br)
-	br = b.putBalances[creator]
-	a.Equal(basics.AppParams{}, br.AppParams[appIdx])
-	a.Equal(basics.StateSchema{}, br.TotalAppSchema)
-	a.Equal(basics.EvalDelta{}, ad.EvalDelta)
-	a.Equal(uint32(0), br.TotalExtraAppPages)
+	cp = basics.AccountData{
+		AppParams:          map[basics.AppIndex]basics.AppParams{appIdx: params},
+		TotalExtraAppPages: 1,
+	}
+	b.balances[creator] = cp
+
+	// check deletion
+	for expTotalExtraPages := uint32(0); expTotalExtraPages < 3; expTotalExtraPages++ {
+		cbr = basics.AccountData{
+			AppParams:          map[basics.AppIndex]basics.AppParams{appIdx: params},
+			TotalExtraAppPages: expTotalExtraPages,
+		}
+		cp := basics.AccountData{
+			AppParams:          map[basics.AppIndex]basics.AppParams{appIdx: params},
+			TotalExtraAppPages: expTotalExtraPages,
+		}
+		b.balances[creator] = cp
+		b.pass = true
+		b.balances[sender] = basics.AccountData{}
+		err = ApplicationCall(ac, h, &b, ad, &ep, txnCounter)
+		a.NoError(err)
+		a.Equal(appIdx, b.deAllocatedAppIdx)
+		a.Equal(1, b.put)
+		br = b.balances[creator]
+		a.Equal(cbr, br)
+		br = b.putBalances[creator]
+		a.Equal(basics.AppParams{}, br.AppParams[appIdx])
+		a.Equal(basics.StateSchema{}, br.TotalAppSchema)
+		a.Equal(basics.EvalDelta{}, ad.EvalDelta)
+		if expTotalExtraPages <= params.ExtraProgramPages {
+			a.Equal(uint32(0), br.TotalExtraAppPages)
+		} else {
+			a.Equal(expTotalExtraPages-1, br.TotalExtraAppPages)
```

Comment:
I'm not following why br.TotalExtraAppPages==expTotalExtraPages-1. isn't it set to expTotalExtraPages on line 1116?

Reply:
`expTotalExtraPages` is a poor name, it should be `initTotalExtraPages`